### PR TITLE
Get sdk integration tests working with Spark

### PR DESCRIPTION
**Pull request by hsubbaraj-spiral**
## Describe your changes and why you are making these changes
Modifies tests to work with Spark. Most of the changes involve adding a case in the test functions to use Pyspark Dataframe specific code. I also add the `pytest.mark.skip_for_spark_engines() `fixture to skip tests that don't work with Spark.
## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: integration_tests/sdk/aqueduct_tests/test_functions/simple/model.py [hunk 1]
```diff
@@ -1,31 +1,64 @@
...
 from aqueduct import op
 
 
 @op()
 def dummy_sentiment_model(df):
-    df["positivity"] = 123
+    if isinstance(df, pd.DataFrame):
+        df["positivity"] = 123
+    else:
+        from pyspark.sql.functions import lit
```
Comment (kenxu95) [inline integration_tests/sdk/aqueduct_tests/test_functions/simple/model.py:11]: Could you put a banner comment on top of this file explaining what the `lit` import does in the spark case? Not clear to me why it has to be a decimal value either.

File: src/golang/lib/job/spark.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ func NewSparkJobManager(conf *SparkJobManagerConfig) (*SparkJobManager, error) {
 
 	session, err := livyClient.CreateSession(&spark.CreateSessionRequest{
 		Kind:                     "pyspark",
-		HeartbeatTimeoutInSecond: 600,
+		HeartbeatTimeoutInSecond: 10,
```
Comment (kenxu95) [inline src/golang/lib/job/spark.go:30]: I assume this was a necessary change to get it working with the test suite? This seems like quite a large numeric change - what are the ramifications of this?

**Review by kenxu95 — CHANGES_REQUESTED** [review]
Just a couple things I want to check on!

File: integration_tests/sdk/aqueduct_tests/checks_test.py [hunk 3]
```diff
@@ -165,15 +175,22 @@ def test_check_wrong_number_of_inputs(client, data_integration):
 
     # TODO(ENG-863): Do we want a more specific error here?
     with pytest.raises(AqueductError):
-        success_on_single_table_input(table_artifact1, table_artifact2)
+        check_artifact = success_on_single_table_input(table_artifact1, table_artifact2)
+        check_artifact.get()
```
[the remark this answers is not in the document]
Comment (hsubbaraj-spiral) [reply integration_tests/sdk/aqueduct_tests/checks_test.py:179]: Yep this is correct!

File: integration_tests/sdk/aqueduct_tests/checks_test.py [hunk 4]
```diff
@@ -165,15 +175,22 @@ def test_check_wrong_number_of_inputs(client, data_integration):
...
 
 
+@pytest.mark.skip
```
[the remark this answers is not in the document]
Comment (hsubbaraj-spiral) [reply integration_tests/sdk/aqueduct_tests/checks_test.py:182]: Yes, should be converted to `@pytest.mark.skip_for_spark_engines()`

File: integration_tests/sdk/conftest.py [hunk 5]
```diff
@@ -207,6 +207,21 @@ def test_k8s(engine):
...
+    (Databricks or Spark)
+    """
+    if request.node.get_closest_marker("skip_for_spark_engines"):
+        if engine and _type_from_engine_name(client, engine) in [
+            ServiceType.DATABRICKS,
+            ServiceType.SPARK,
+        ]:
+            pytest.skip(
+                "Skipped for engine integration `%s`, since it is a spark-based engine." % engine
+            )
```
[the remark this answers is not in the document]
Comment (hsubbaraj-spiral) [reply integration_tests/sdk/conftest.py:222]: Ah this makes sense, was wondering how to hide the errors.

File: integration_tests/sdk/aqueduct_tests/flow_test.py [hunk 6]
```diff
@@ -384,6 +385,7 @@ def test_multiple_flows_with_same_schedule(client, flow_name, data_integration,
     )
 
 
+@pytest.mark.skip_for_spark_engines(reason="requires implicit pandas requirement.")
```
Comment (kenxu95) [inline integration_tests/sdk/aqueduct_tests/flow_test.py:388]: my weird pet peeve/nit: Capitalize "requires"

**Review by kenxu95 — APPROVED** [review]

**Pull request merged**